src/lib/services/user-service.ts

```diff
@@ -273,8 +275,8 @@ class UserService {
             simpleAuthKey,
         );
 
-        if (settings && settings.disabled) {
-            throw new Error(
+        if (settings?.disabled) {
+            throw new DisabledError(
```

Comment:
The DisabledError could be used for any kind of feature that is currently disabled. By encoding the status code in to the Error type itself we do not have to do more than simply throw it when the user hits a feature they do not have access to currently, and it will automatically map to a nice HTTP 422 response.